### PR TITLE
Vctr 55 implement clamp, clamp low, clamp high

### LINKED ISSUE #29: Implement clamp, clampLow, clampHigh

### PR DESCRIPTION
**Pull request by JanosGit**
This includes a bit of rework of the `ExpressionChainBuilder` class in order to make it possible to pass variables computed at runtime to expressions.

I'd recommend going over the Expressions/Readme additions first, then have a look at the changes of the ExpressionChainBuilder and finally check the implementation of the clamp functions

### CODE REVIEW

File: include/vctr/Expressions/Core/Clamp.h [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+constexpr auto clamp (T lowerBound, T upperBound)
+{
+    return makeExpressionChainBuilderWithRuntimeArgs<Expressions::ClampLowHigh> (lowerBound, upperBound);
+}
+
+
+
+/** Ensures that the elements are not lower than lowerBound.
+
+    Note: At this time, most compilers wil refuse to take floating point non-type template values.
```
Comment (TheWaveWarden) [inline include/vctr/Expressions/Core/Clamp.h]: *will refuse. Also two instances below 🙂

File: include/vctr/Expressions/ExpressionChainBuilder.h [hunk 2]
```diff
@@ -25,13 +25,68 @@ namespace vctr
 
 namespace detail
 {
+/** Internal helper structure to pass runtime arguments to expressions in a chain
+    built from a concatenation of expression chain builder instances.
+
+    The expression chain builder primarily holds a template template for the
```
Comment (TheWaveWarden) [inline include/vctr/Expressions/ExpressionChainBuilder.h:31]: a template template? 😄

**Review by TheWaveWarden — APPROVED** [review]
Looks good 👍

File: include/vctr/Expressions/ExpressionChainBuilder.h (hunk 2 above)
Comment (JanosGit) [reply include/vctr/Expressions/ExpressionChainBuilder.h:31]: Yes, indeed "template template" is an term used by the community, see e.g. the section on it on [cpprefence](https://en.cppreference.com/w/cpp/language/template_parameters)
Screenshot 2023-01-10 at 11 49 59

File: include/vctr/Expressions/Core/Clamp.h (hunk 1 above)
Comment (JanosGit) [reply include/vctr/Expressions/Core/Clamp.h]: Ah yes, will fix that

**Pull request merged**